webview/icons/generate.mjs

```diff
@@ -0,0 +1,103 @@
+import { transform } from '@svgr/core'
+import {
+  appendFile,
+  mkdir,
+  readdir,
+  readFile,
+  rm,
+  writeFile
+} from 'node:fs/promises'
+import path from 'path'
+
+const iconsPath = 'src/shared/components/icons/'
+
+try {
+  await rm(iconsPath, { recursive: true, force: true })
+  await mkdir(iconsPath)
+} catch (err) {
+  console.error(err.message)
+}
+
+const customIcons = []
+try {
+  const files = await readdir('icons')
+  for (const file of files) {
+    if (path.extname(file) === '.svg') {
+      customIcons.push(`icons/${file}`)
+    }
+  }
+} catch (err) {
+  console.error(err.message)
+}
+
+const codiconsPath = '../node_modules/@vscode/codicons/src/icons'
+const codicons = [
+  'add',
+  'arrow-up',
+  'arrow-down',
+  'beaker',
+  'check',
+  'chevron-down',
+  'chevron-right',
+  'close',
+  'copy',
+  'ellipsis',
+  'error',
+  'filter',
+  'git-commit',
+  'git-merge',
+  'graph-line',
+  'graph-scatter',
+  'gripper',
+  'info',
+  'list-filter',
+  'pass-filled',
+  'pinned',
+  'refresh',
+  'sort-precedence',
+  'star-empty',
+  'star-full',
+  'trash'
+]
+
+const allIcons = [
+  ...customIcons,
+  ...codicons.map(codicon => `${codiconsPath}/${codicon}.svg`)
+]
+
+const toPascalCase = string =>
```

Comment:
If I'm understanding this code correctly, it's converting a kebab-case word to a PascalCase word? If so, I found this [simpler solution on SlackOverflow](https://stackoverflow.com/a/54651317) a while back:

```
function toPascalCase(text) {
  return text.replace(/(^\w|-\w)/g, clearAndUpper);
}

function clearAndUpper(text) {
  return text.replace(/-/, "").toUpperCase();
}
```

Reply:
Nice! I just grabbed the first thing I saw. This looks better.